pkg/idp/validation.go

```diff
@@ -23,9 +23,22 @@ type PayloadValidator struct {
 	logger logging.LoggerInterface
 }
 
+func genericIssuerOAuth2URLsValidation(sl validator.StructLevel) {
+	configuration := sl.Current().Interface().(Configuration)
+
+	if configuration.Provider == "generic" {
```

Comment:
Just a minor nitpick (feel free to ignore):

We could return early to do one less `if` block:

```go
if configuration.Provider != "generic" {
    return
}

if configuration.IssuerURL == "" && (configuration.AuthURL == "" || configuration.TokenURL == "") {
		sl.ReportError(configuration.IssuerURL, "issuer_url", "IssuerURL", "issuer_urls", "")
}
```

Reply:
Done :) Thanks for the suggestion! I pushed again with the change